src/Update/PowerToys.Update.cpp

```diff
@@ -185,13 +169,33 @@ bool InstallNewVersionStage2(std::wstring installer_path, std::wstring_view inst
         }
     }
 
+    for (const auto& entry : fs::directory_iterator(updating::get_pending_updates_path()))
+    {
+        auto entryPath = entry.path().wstring();
+        std::transform(entryPath.begin(), entryPath.end(), entryPath.begin(), ::towlower);
+
+        if ((entryPath.ends_with(L".msi") || entryPath.ends_with(L".exe")) && installer_path.find(entryPath) == std::string::npos)
+        {
+            std::error_code err;
+            fs::remove(entry, err);
+            if (err.value())
+            {
+                Logger::warn("Failed to delete file {}. {}", entry.path().string(), err.message());
+            }
+        }
+    }
+
     if (!success)
     {
         return false;
     }
 
-    std::error_code _;
-    fs::remove(installer_path, _);
+    std::error_code err;
+    fs::remove(installer_path, err);
```

Comment:
Does this make sense? If we delete the old files after every update, this will be deleted after the next update. And btw why don't we delete the installer of the current update in line 180 if the update is success? This seems to be obsolete.